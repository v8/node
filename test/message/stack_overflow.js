'use strict';
require('../common');

Error.stackTraceLimit = 0;

console.error('before');

// Trigger stack overflow by stringifying a deeply nested array.
var array = [];
for (var i = 0; i < 100000; i++) {
  array = [ array ];
}

JSON.stringify(array);

console.error('after');
