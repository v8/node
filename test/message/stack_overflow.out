before
/usr/local/google/home/yangguo/node/test/message/stack_overflow.js:14
JSON.stringify(array);
     ^

RangeError: Maximum call stack size exceeded
